src/main/features/claim/helpers/featuresBuilder.ts

```diff
@@ -19,7 +22,7 @@ export class FeaturesBuilder {
 
   async features (amount: number, user: User): Promise<string> {
     const roles: string[] = await this.claimStoreClient.retrieveUserRoles(user)
-    if (!roles.includes('cmc-new-features-consent-given')) {
+    if (!featureToggles.isAutoEnrollIntoNewFeatureEnabled() && !roles.includes('cmc-new-features-consent-given')) {
```

Comment:
This should be an async function

Reply:
Acknowledged. Necessary changes are made as we have discussed.